### DISCUSSION TITLE
Feature request: Displaying collection field choices

### DISCUSSION THREAD

**Discussion by titomitto** [pocketbase/pocketbase#1107] [Q&A] [answered]
I love this project to death, but one of the pain points I have discovered when working with collections is that when I add a field with choices there is really no way to know from the front-end (JS or Dart ) which choices are present in the collection. It would be nice if someone could know the choices for specific fields, otherwise what someone has to do currently is to hardcode choices on the front-end also.
I have two ideas of how this could be approached. For example, if you had `artworks` collection you would query the accepted choices like this `client.collections("artworks").listChoicesForField("type")`. Another alternative would be to query all fields that have choices `client.collections("artworks").listAllFieldsWithChoices()`. Is this something that is currently possible? It would be nice to have.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure about this. Creating a dedicated endpoint and making request to return a single collection field options seems ambiguous and excessive. What would be the result of the above if the field is not a `select` but a `relation` for example?

In any case, I don't think this is needed because you still will have to map the options somewhere in your frontend in order to show a human-readable label, icon, etc.

**Reply by titomitto** [reply]
The issue I actually had with it was that I wanted the fields to be dynamically set so that I don't have to release another app version every time there is a new type added to the collection. But it actually does make sense that I should use another collection for cases where I may need to add more options.

**Comment by denisgurev** [comment]
I think that the multiselect options are more suitable for a limited number of choices like status or type, where most likely you'll have 4-5 values to choose from.

If you have more options then maybe replacing the select with a relation field to a separate collection will probably be better in this case. I do something similar by keeping my "country" options in a "countries" collection. This allow me to query the "countries" collection and I can also define additional fields for each country/option like title or alias for example.
